chandra_aca/maude_decom.py

```diff
@@ -0,0 +1,451 @@
+        | G1 | O1 | G2 | O2 | G3 | O3 | G4 | O4 |
+        -----------------------------------------
+        | F1 | N1 | F2 | N2 | F3 | N3 | F4 | N4 |
+        -----------------------------------------
+        | E1 | M1 | E2 | M2 | E3 | M3 | E4 | M4 |
+        -----------------------------------------
+        | D1 | L1 | D2 | L2 | D3 | L3 | D4 | L4 |
+        -----------------------------------------
+        | C1 | K1 | C2 | K2 | C3 | K3 | C4 | K4 |
+        -----------------------------------------
+        | B1 | J1 | B2 | J2 | B3 | J3 | B4 | J4 |
+        -----------------------------------------
+        | A1 | I1 | A2 | I2 | A3 | I3 | A4 | I4 |
+        -----------------------------------------
+
+    NOTE: in the previous tables, the rows are numbered in _ascending_ order.
+    If one prints the corresponding np.array they are printed in _descending_ order,
+    while if you draw them using plt.pcolor they will be in ascending order.
+
+    :param pixel_data: dictionary containing MSID values and times.
+
+    The keys of ``pizel_data`` must be MSIDs.
+    The values must be dictionaries with keys ['times', 'values']
+
+    :param img_size: an array of image size specifications.
+
+    Each entry in ``img_size`` must be one of: 4X41, 6X61, 6X62, 8X81, 8X82, 8X83 or 8X84.
+    The size of ``img_size`` must be equal to the size of the last axis of ``pixel_data``.
+
+    """
+    if list(pixel_data.values())[0]['values'].shape[-1] != len(img_size):
+        s1, s2 = list(pixel_data.values())[0]['values'].shape[-1], len(img_size)
+        raise Exception(
+            f'Pixel data shape ({s1},) and image size array shape ({s2},) do not agree.')
+
+    img = np.ones((len(img_size), 8, 8)) * np.nan
+
+    msid_img = list(set([k[:-2] for k in pixel_data.keys()]))
+    assert len(msid_img) == 1
+    msid_img = msid_img[0]
+
+    size_4x4 = (img_size == '4X41')
+    size_6x6 = (img_size == '6X61') + (img_size == '6X62')
+    size_8x8 = ((img_size == '8X81') + (img_size == '8X82') +
+                (img_size == '8X83') + (img_size == '8X84'))
+
+    # -----------------------------------------
+    # | -- | -- | -- | -- | -- | -- | -- | -- |
+    # -----------------------------------------
+    # | -- | -- | -- | -- | -- | -- | -- | -- |
+    # -----------------------------------------
+    # | -- | -- | D1 | H1 | L1 | P1 | -- | -- |
+    # -----------------------------------------
+    # | -- | -- | C1 | G1 | K1 | O1 | -- | -- |
+    # -----------------------------------------
+    # | -- | -- | B1 | F1 | J1 | N1 | -- | -- |
+    # -----------------------------------------
+    # | -- | -- | A1 | E1 | I1 | M1 | -- | -- |
+    # -----------------------------------------
+    # | -- | -- | -- | -- | -- | -- | -- | -- |
+    # -----------------------------------------
+    # | -- | -- | -- | -- | -- | -- | -- | -- |
+    # -----------------------------------------
+    img[size_4x4, 2, 2] = pixel_data[f'{msid_img}A1']['values'][size_4x4]
```

Comment:
The comments are good and the assignments seem fine, but I'm wondering if the source data for the assignments can just *be* a Table with the pixel labels, like

https://github.com/sot/chandra_aca/blob/24283eed62f1089947b5d9b5008f36051b7bf16c/chandra_aca/aca_image.py#L17